packages/stt-adapters/digital-paper-edit/group-words-by-speakers.js

```diff
@@ -0,0 +1,102 @@
+/**
+edge cases
+- more segments then words - not an issue if you start by matching words with segment
+and handle edge case where it doesn't find a match
+- more words then segments - orphan words
+ */
+function groupWordsInParagraphsBySpeakers(words, segments) {
+  // add speakers to each word
+  const wordsWithSpeakers = addSpeakerToEachWord(words, segments);
+  // group words by speakers sequentially
+  const result = groupWordsBySpeaker(wordsWithSpeakers);
+
+  return result;
+};
+
+/**
+* Add speakers to each words
+* if it doesn't have add unknown attribute `U_UKN`
+* @param {*} words
+* @param {*} segments
+*/
+function addSpeakerToEachWord(words, segments) {
+  const tmpWordsWithSpeakers = [];
+  words.forEach((word) => {
+    const tmpSpeaker = findSegmentForWord(word, segments);
+
+    word.speaker = tmpSpeaker;
```

Comment:
It's good practice here to create a new `word` object rather than modify the original one. This would in term modify the whole words array passed in (perhaps not a big deal in this function...but can cause unexpected behaviours elsewhere).

`Object.assign()` might be what we want here.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

Reply:
I think it's trailing some logic from one of the other adapters. 

This might also be enough?

```js
word.speaker = findSegmentForWord(word, segments);
```

Unless I am missing something (@jamesdools )

Reply:
refactored to be 

```js
function addSpeakerToEachWord(words, segments) {
  return words.map((word) => {
    word.speaker = findSegmentForWord(word, segments);

    return word;
  });
}
```